metadata/memory_lookup.go

```diff
@@ -0,0 +1,333 @@
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/featureform/fferr"
+	"github.com/featureform/logging"
+	pb "github.com/featureform/metadata/proto"
+	"github.com/featureform/storage"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+// Create Resource Lookup Using ETCD
+type MemoryResourceLookup struct {
+	Connection storage.MetadataStorage
+}
+
+// Wrapper around Resource/Job messages. Allows top level storage for info about saved value
+type MemoryRow struct {
+	ResourceType ResourceType //Resource Type. For use when getting stored keys
+	//ResourceType string
+	StorageType StorageType //Type of storage. Resource or Job
+	Message     []byte      //Contents to be stored
+}
+
+type MemoryRowTemp struct {
+	//ResourceType ResourceType //Resource Type. For use when getting stored keys
+	ResourceType ResourceType
+	StorageType  StorageType //Type of storage. Resource or Job
+	Message      []byte      //Contents to be stored
+}
+
+// Takes a populated ETCD Storage struct and a resource
+// Checks to make sure the given ETCD Storage Object contains a Resource, not job
+// Deserializes Resource value into the provided Resource object
+func (s MemoryResourceLookup) ParseResource(res EtcdRow, resType Resource) (Resource, error) {
+	if res.StorageType != RESOURCE {
+		return nil, fmt.Errorf("payload is not resource type")
+	}
+
+	if !resType.Proto().ProtoReflect().IsValid() {
+		return nil, fmt.Errorf("cannot parse to invalid resource")
+	}
+
+	if res.Message == nil {
+		return nil, fmt.Errorf("cannot parse invalid message")
+	}
+
+	if err := proto.Unmarshal(res.Message, resType.Proto()); err != nil {
+		return nil, err
+	}
+
+	return resType, nil
+}
+
+// Returns an empty Resource Object of the given type to unmarshal etcd value into
+func (lookup MemoryResourceLookup) createEmptyResource(t ResourceType) (Resource, error) {
+	var resource Resource
+	switch t {
+	case FEATURE:
+		resource = &featureResource{&pb.Feature{}}
+		break
+	case FEATURE_VARIANT:
+		resource = &featureVariantResource{&pb.FeatureVariant{}}
+		break
+	case LABEL:
+		resource = &labelResource{&pb.Label{}}
+		break
+	case LABEL_VARIANT:
+		resource = &labelVariantResource{&pb.LabelVariant{}}
+		break
+	case USER:
+		resource = &userResource{&pb.User{}}
+		break
+	case ENTITY:
+		resource = &entityResource{&pb.Entity{}}
+		break
+	case PROVIDER:
+		resource = &providerResource{&pb.Provider{}}
+		break
+	case SOURCE:
+		resource = &SourceResource{&pb.Source{}}
+		break
+	case SOURCE_VARIANT:
+		resource = &sourceVariantResource{&pb.SourceVariant{}}
+		break
+	case TRAINING_SET:
+		resource = &trainingSetResource{&pb.TrainingSet{}}
+		break
+	case TRAINING_SET_VARIANT:
+		resource = &trainingSetVariantResource{&pb.TrainingSetVariant{}}
+		break
+	case MODEL:
+		resource = &modelResource{&pb.Model{}}
+		break
+	default:
+		return nil, fmt.Errorf("Invalid Type\n")
```

Comment:
Would make this error, Unable to create empty resource of type X. Unimplemeted. And make it an internal error

Reply:
Change and moved